### PR TITLE
etcd_worker (ticdc): fix capture restart since owner resgin

### LINKED ISSUE #7398: resign the owner cause the capture exit
### What did you do?

`tiup cluster reload` the cluster to trigger owner switch

### What did you expect to see?

owner capture resign the ownership, and the `capture` goroutine works normally.

### What did you see instead?

```
[2022/10/18 22:05:28.352 +08:00] [INFO] [owner.go:666] ["cleanup owner jobs as owner has been closed"]
[2022/10/18 22:05:28.352 +08:00] [INFO] [client.go:235] ["WatchWithChan exited"] [role=owner]
[2022/10/18 22:05:28.353 +08:00] [INFO] [capture.go:467] ["owner resigned successfully"] [captureID=9d38a6d1-3910-477b-a921-8f0917bb347f] [ownerRev=10915858]
[2022/10/18 22:05:28.353 +08:00] [INFO] [capture.go:477] ["run owner exited normally"] [captureID=9d38a6d1-3910-477b-a921-8f0917bb347f] [ownerRev=10915858]
[2022/10/18 22:05:28.354 +08:00] [ERROR] [etcd_worker.go:356] ["owner key is deleted externally, exit etcd worker and suicide the capture"] [ownerMetaKey=/tidb/cdc/default/__cdc_meta__/owner/722b83e4cc26dcdc] [value=]
[2022/10/18 22:05:28.354 +08:00] [INFO] [client.go:235] ["WatchWithChan exited"] [role=processor]
[2022/10/18 22:05:28.354 +08:00] [INFO] [capture.go:337] ["processor routine exited"] [captureID=9d38a6d1-3910-477b-a921-8f0917bb347f] [error="[CDC:ErrCaptureSuicide]capture suicide"] [errorVerbose="[CDC:ErrCaptureSuicide]capture suicide\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).handleEvent\n\tgithub.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:360\ngithub.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run\n\tgithub.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:230\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:494\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:336\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=236] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=295] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=709] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=587] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=272] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=777] []
[2022/10/18 22:05:28.355 +08:00] [WARN] [reader.go:427] ["receive stop message"] [tableID=799] []
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7398

#### Tests 

 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/capture/capture.go [hunk 1]
```diff
@@ -311,11 +311,12 @@ func (c *captureImpl) run(stdCtx context.Context) error {
 	g.Go(func() error {
 		// when the campaignOwner returns an error, it means that the owner throws
 		// an unrecoverable serious errors (recoverable errors are intercepted in the owner tick)
-		// so we should also stop the owner and let capture restart or exit
+		// so we should also stop the owner and let capture **restart**
 		err := c.campaignOwner(ctx)
-		log.Info("owner routine exited",
+		log.Error("owner routine exited",
```
Comment (3AceShowHand) [inline cdc/capture/capture.go:316]: If the owner is resigned and exit normally, it should be `Error`

**Review by 3AceShowHand — APPROVED** [review]

File: pkg/orchestrator/etcd_worker.go [hunk 2]
```diff
@@ -229,7 +230,9 @@ func (worker *EtcdWorker) Run(ctx context.Context, session *concurrency.Session,
 				// handleEvent will apply the event to our internal `rawState`.
 				err = worker.handleEvent(ctx, event)
 				if err != nil {
-					return err
+					// This error means owner is resigned by itself,
+					// and we should exit etcd worker and campaign owner again.
```
Comment (sdojjy) [inline pkg/orchestrator/etcd_worker.go:234]: log the error

Comment (asddongmen) [reply pkg/orchestrator/etcd_worker.go:234]: This error is log in the `handleEvent` func.

**Review by sdojjy — APPROVED** [review]

**Pull request merged**